web/html/src/manager/notifications/notifications-list.tsx

```diff
@@ -0,0 +1,454 @@
+import * as React from "react";
+
+import { AsyncButton, LinkButton } from "components/buttons";
+import { Dialog } from "components/dialog/Dialog";
+import { Form, Select } from "components/input";
+import { Messages as MessageContainer, MessageType, Utils as MessagesUtils } from "components/messages/messages";
+import { TopPanel } from "components/panels/TopPanel";
+import { SectionToolbar } from "components/section-toolbar/section-toolbar";
+import { Column } from "components/table/Column";
+import { SearchField } from "components/table/SearchField";
+import { Table } from "components/table/Table";
+
+import { stringToReact } from "utils";
+import { Utils } from "utils/functions";
+import Network from "utils/network";
+
+export type NotificationType = {
+  label: string;
+  description: string;
+};
+
+enum Severity {
+  Info = "info",
+  Warning = "warning",
+  Error = "error",
+}
+
+type Notification = {
+  id: number;
+  severity: Severity;
+  type: string;
+  summary: string;
+  details?: string;
+  read: boolean;
+  actionable: boolean;
+  create: Date;
+};
+
+type Props = {
+  notificationTypes: Array<NotificationType>;
+};
+
+type State = {
+  serverData: Array<Notification> | undefined;
+  dataUrlTags: Array<string>;
+  currentDataUrlTag: string;
+  loading: boolean;
+  messages: Array<MessageType>;
+  selectedItems: Array<number>;
+  popupItem: Notification | undefined;
+  typeCriteria: Array<string>;
+};
+
+export class NotificationList extends React.Component<Props, State> {
+  private readonly typeMap: Map<string, string>;
+
+  private readonly sortedNotificationTypes: Array<NotificationType>;
+
+  public constructor(props: Props) {
+    super(props);
+
+    // Sort the notification types by description
+    this.sortedNotificationTypes = [...this.props.notificationTypes].sort((a, b) =>
```

Comment:
Since these values are coming straight from props, is there a reason not to send them in the right order to begin with?

Reply:
Fixed.